force-app/src/classes/MethodSpy.cls

```diff
@@ -53,40 +52,51 @@ global class MethodSpy {
   public Object call(List<Type> paramTypes, List<String> paramNames, List<Object> args) {
     this.callLog.add(new MethodCall(args));
 
-    if (this.parameterizedMethodCalls.isEmpty() && !this.configuredGlobalReturn && !this.configuredGlobalThrow) {
+    if (!this.isConfigured()) {
       return null;
     }
 
-    for (ParameterizedMethodSpyCall parameterizedCall : this.parameterizedMethodCalls) {
-      if (parameterizedCall.matches(args)) {
-        if (parameterizedCall.shouldThrow()) {
-          throw parameterizedCall.error;
-        }
-        return parameterizedCall.value;
+    for (ParameterizedMethodSpyCall methodCall : this.methodCalls) {
+      if (methodCall.matchesOnce(args)) {
+        return methodCall.getConfiguredBehavior();
```

Comment:
suggestion: change `getConfiguredBehavior` to something more explicitly saying that a side-effect can remove a behavior configured "once"